experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/proxyNodeCursor.ts

```diff
@@ -0,0 +1,166 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/core-utils";
+import { type Value, type FieldKey, EmptyKey, TreeTypeSet, rootFieldKey } from "../../../core";
+// eslint-disable-next-line import/no-internal-modules
+import { leaf } from "../../../domains/leafDomain";
+import {
+	ContextuallyTypedNodeData,
+	TreeDataContext,
+	getPossibleTypes,
+	isFluidHandle,
+	isPrimitiveValue,
+} from "../../contextuallyTyped";
+import {
+	type CursorAdapter,
+	type CursorWithNode,
+	stackTreeNodeCursor,
+	stackTreeFieldCursor,
+} from "../../treeCursorUtils";
+import { type TreeNodeSchema } from "../../typed-schema";
+import { type ProxyNode } from "./types";
+
+/**
+ * Creates a {@link CursorAdaptor} capable of mapping proxy-domain trees to the editable tree domain.
+ *
+ * @param context - The {@link TreeEntity.context} of the parent node under which the tree is being inserted.
+ * @param typeSet - The types allowed under the parent node under which the tree is being inserted.
+ *
+ * @privateRemarks
+ * This is exported solely for testing purposes.
+ * Consumers should use {@link cursorFromProxyTreeNode} instead.
+ */
+export function createAdaptor<TNode extends ProxyNode<TreeNodeSchema, "javaScript">>(
+	context: TreeDataContext,
+	typeSet: TreeTypeSet,
+): CursorAdapter<TNode> {
+	return {
+		value: (node: TNode) => {
+			return isPrimitiveValue(node) || node === null || isFluidHandle(node)
+				? (node as Value)
+				: undefined;
+		},
+		type: (node: TNode) => {
+			if (node === undefined) {
+				throw new Error("undefined node");
+			}
+			if (node === null) {
+				return leaf.null.name;
+			}
+
+			if (isFluidHandle(node)) {
```

Comment:
Can we move this into the `default` part of the switch statement? It's slower than checking `number`, `string`, or `boolean`, so it's a little micro-op to have it be checked after those. I don't want to _prematurely_ micro-op, but in this case it doesn't have any code complexity cost to move it there.